packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -142,6 +145,37 @@ export class ProjectDataSource {
     return matched
   }
 
+  startSpecWatcher (projectRoot: string, testingType: Cypress.TestingType, specPattern: string | string[]) {
+    if (this._specWatcher) {
+      this.stopSpecWatcher()
+    }
+
+    const currentProject = this.ctx.currentProject
+
+    if (!currentProject) {
+      throw new Error('Cannot start spec watcher without current project')
+    }
+
+    const onSpecsChanged = debounce(async () => {
+      const specs = await this.findSpecs(projectRoot, testingType, specPattern)
+
+      this.setSpecs(specs)
+      if (testingType === 'component') {
+        this.api.getDevServer().updateSpecs(specs)
+      }
+
+      this.ctx.emitter.toApp()
+    })
+
+    this._specWatcher = this.ctx.lifecycleManager.addWatcher(specPattern)
+    this._specWatcher.on('add', onSpecsChanged)
+    this._specWatcher.on('unlink', onSpecsChanged)
+  }
+
+  stopSpecWatcher () {
```

Comment:
Should this just be 

```ts
if (!this._specWatcher) {
  return
}

this.ctx.lifecycleManager.closeWatcher(this._specWatcher)
```

Then we do not need to loop over the watchers inside of `closeWatcher` and do the null comparison - no need to call the method if the watcher is null.

Reply:
Makes sense!